docs/docs/forms.mdx

```diff
@@ -388,6 +388,35 @@ class ValidateRestaurantForm(FormValidationAction):
         return additional_slots + domain_slots
 ```
 
+If conversely, you'd prefer to remove a slot from the form's `required_slots` defined in the domain file in certain conditions,
+we recommend to copy the `domain_slots` over to a new variable and applying changes to that new variable instead of modifying directly
```

Comment:
```suggestion
you can copy the `domain_slots` over to a new variable and apply changes to that new variable instead of directly modifying
```

Reply:
will anything bad happen if they do modify it directly or is it just for clarity?

Reply:
They won't get the behaviour they want (not requesting the removed slot), because Rasa will instead use the list of required slots defined in the domain, I explained how this occurs in the comment [here](https://github.com/RasaHQ/rasa/issues/10536#issuecomment-1020405414).

Reply:
Ah I see thanks! 
In that case we should be stronger in the wording e.g.
```
you should copy the `domain_slots` over to a new variable and apply changes to that new variable instead of directly modifying `domain_slots`. Directly modifying `domain_slots` can cause unexpected behaviour.
```